cranelift/codegen/src/isa/x64/lower.isle

```diff
@@ -1501,6 +1501,38 @@
 (rule (lower (icmp cc a @ (value_type $I128) b))
       (lower_icmp_bool (emit_cmp cc a b)))
 
+;; Peephole optimization for `x < 0`, when x is a signed 64 bit value
+(rule (lower (icmp (IntCC.SignedLessThan) x @ (value_type $I64) (u64_from_iconst 0)))
+      (x64_sar $I64 x (Imm8Reg.Imm8 63)))
+
+;; Peephole optimization for `0 > x`, when x is a signed 64 bit value
+(rule (lower (icmp (IntCC.SignedGreaterThan) (u64_from_iconst 0) x @ (value_type $I64)))
+      (x64_sar $I64 x (Imm8Reg.Imm8 63)))
```

Comment:
Doesn't ISLE and cranelift ir always canonicalize constants on the right?

Reply:
Not currently, hence the symmetric cases for this optimization. This is something we might look into after the mid-end optimization work has landed.

Reply:
Got it. Thanks!